lib/questions.rb

```diff
@@ -1,226 +1,295 @@
+
+  # round up a float up and convert it to an Integer,
+  # so 3.214 becomes 4
+  def round_up_number(float)
+    float.ceil
+  end
+
+  # round down a float up and convert it to an Integer,
+  # so 9.52 becomes 9
+  def round_down_number(float)
+    float.floor
+  end
+
+  # take a date and format it like dd/mm/yyyy, so Halloween 2013
+  # becomes 31/10/2013
+  def format_date_nicely(date)
+    date.strftime("%d/%m/%Y")
+  end
+
+  # get the domain name *without* the .com part, from an email address
+  # so alex@makersacademy.com becomes makersacademy
+  def get_domain_name_from_email_address(email)
+    domain = email.gsub(/.+@([^.]+).+/, '\1')
+  end
+
+  # capitalize the first letter in each word of a string,
+  #  except 'a', 'and' and 'the'
+  # *unless* they come at the start of the start of the string, e.g.
+  # 'the lion the witch and the wardrobe' becomes
+  # 'The Lion the Witch and the Wardrobe'
+  def titleize_a_string(string)
+    title_cases = Hash.new {|_,k| k.capitalize }.merge({'And' => 'and', 'The' => 'the', 'A' => 'a'})
+    new_string  = string.split(" ").map {|w| title_cases[w] }.join(' ')
+    new_string.capitalize
+  end
+
+  # return true if a string contains any special characters
+  # where 'special character' means anything apart from the letters
+  # a-z (uppercase and lower) or numbers
+  def check_a_string_for_special_characters(string)
+    /\W/ === string
+  end
+
+  # get the upper limit of a range. e.g. for the range 1..20, you
+  # should return 20
+  def get_upper_limit_of(range)
+    range.last
+  end
+
+  # should return true for a 3 dot range like 1...20, false for a
+  # normal 2 dot range
+  def is_a_3_dot_range?(range)
+    range.exclude_end?
+  end
+
+  # get the square root of a number
+  def square_root_of(number)
+    Math.sqrt(number)
+  end
+
+  # count the number of words in a file
+  def word_count_a_file(file_path)
+  end
+
+  # --- tougher ones ---
+
+  # call an arbitrary method from a string. so if I
+  # called call_method_from_string('foobar')
+  # the method foobar should be invoked
+  def call_method_from_string(str_method)
+  end
+
+  # return true if the date is a uk bank holiday for 2014
+  # the list of bank holidays is here:
+  # https://www.gov.uk/bank-holidays
+  def is_a_2014_bank_holiday?(date)
+  end
+
+  # given your birthday this year, this method tells you
+  # the next year when your birthday will fall on a friday
+  # e.g. january 1st, will next be a friday in 2016
+  # return the day as a capitalized string like 'Friday'
+  def your_birthday_is_on_a_friday_in_the_year(birthday)
+  end
+
+  # in a file, total the number of times words of different lengths
+  # appear. So in a file with the text "the cat sat on the blue mat"
+  # I have 5 words which are 3 letters long, 1 which is 2 letters long
+  # and 1 that is 4 letters long. Return it as a hash in the format
+  # word_length => count, e.g. {2 => 1, 3 => 5, 4 => 1}
+  def count_words_of_each_length_in_a_file(file_path)
```

Comment:
Unused method argument - `file_path`. If it's necessary, use `_` or `_file_path` as an argument name to indicate that it won't be used. You can also write as `count_words_of_each_length_in_a_file(*)` if you want the method to accept any arguments but don't care about them.